platform-sdk/swirlds-jasperdb/src/test/java/com/swirlds/merkledb/MerkleDbDataSourceMetricsTest.java

```diff
@@ -151,15 +151,17 @@ void createAndCheckLeaves() throws IOException {
                 lastLeafIndex + DEFAULT_RESERVED_BUFFER_LENGTH,
                 lastLeafIndex + DEFAULT_RESERVED_BUFFER_LENGTH + 1,
                 Stream.empty(),
-                IntStream.of(42).mapToObj(i -> fixed_fixed.dataType().createVirtualLeafRecord(i)),
+                // valid leaf index
+                IntStream.of(lastLeafIndex + DEFAULT_RESERVED_BUFFER_LENGTH)
```

Comment:
Previously this test used an index that is below min valid index. With no validation in place it worked but after I added 
`assert index >= minValidIndex.get();` to `chechIndexAndValue` method, it started failing. Now this test uses correct index. And the memory consumption now makes more sense: `offHeapKeyToPath` remains unchanged (not increases) and total memory consumption decreases because `offHeapLeafMb` is lowers to 8.

Reply:
This test made me reconsider where to call `updateMinValidIndex` from, and it looks like `startWriting` makes more sense - if entries are deleted from the data source, we need to update min valid index first to enable writes to the indexes that were below min valid index.